### PR TITLE
feat: change token header to authorization

### PR DESCRIPTION
**Pull request by Graborg**
**Ticket:** [#68](https://trello.com/c/FtcKEySS/68-change-token-name-to-authorization)

**Intent:**
Change token header name to something more appropriate: Authorization

Based on https://github.com/JobtechSwe/myskills-api/pull/36
**How to test :**

- [ ] I've written tests for the code.

### CODE REVIEW

File: lib/adapters/mydata.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ interface Config {
 
 export interface DataInput {
   area: Area
-  token: string
+  token?: string
```
Comment (lundeeell) [inline lib/adapters/mydata.ts]: Token cannot be undefined and must be provided to getData/setData

We should throw an unauthorized error if token is not provided to resolvers that must be authed (everything that reads/writes in mydata)

File: lib/middleware/authorizationToken.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Request, Response, NextFunction } from 'express'
+
+export default (req: Request, _res: Response, next: NextFunction) => {
+  req.token = undefined
+  if (!req.headers.authorization) {
+    return next()
+  }
+  const auth = req.headers.authorization
+  if (!auth.startsWith('Bearer')) {
+    throw Error('Authorization token malformed. Use "Bearer <token>"')
```
Comment (lundeeell) [inline lib/middleware/authorizationToken.ts]: This doesn't work with graphql
"error": "Unexpected token < in JSON at position 0"

Comment (Graborg) [reply lib/middleware/authorizationToken.ts]: Yes, that how we produce all errors at the moment... But I can also add some better error handling. Boyscout style.

Comment (Graborg) [reply lib/middleware/authorizationToken.ts]: Ah. This was easier fixed than I thought. Thanks.

File: lib/adapters/mydata.ts (hunk 1 above)
Comment (Graborg) [reply lib/adapters/mydata.ts]: Yes, thats probably for the best.

**Pull request merged**